src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -461,6 +477,34 @@ private ReduceableSearchResult reduceSearchResults(final List<ReduceableSearchRe
         };
     }
 
+    /**
+     * Get maximum subquery results count to be collected from each shard.
+     * @param searchContext search context that contains pagination depth
+     * @return results size to collected
+     */
+    private static int getSubqueryResultsRetrievalSize(final SearchContext searchContext) {
+        HybridQuery hybridQuery = getHybridQueryFromAbstractQuery(searchContext.query());
+        int paginationDepth = hybridQuery.getPaginationDepth();
+
+        // Switch to from+size retrieval size during standard hybrid query execution.
+        if (searchContext.from() == 0) {
+            return searchContext.from() + searchContext.size();
+        }
+        return paginationDepth;
+    }
+
+    private static HybridQuery getHybridQueryFromAbstractQuery(Query query) {
```

Comment:
can you think of a better method name, it not clear what we're trying to do here